app/models/post.rb

```diff
@@ -1,3 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :author, class_name: 'User'
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def update_user_posts_counter
+    author.update(posts_count: author.posts.count)
+  end
```

Comment:
- [ Required ] I think it would be better to add an active rails callback which ensures that the `posts counter` is updated when a new one is added to the database. This will ensure that any time a post is added to the database `posts _count` is updated, I would like to suggest that you add this line of code below `line 4`

```ruby
  after_save :update_user_posts_counter

  private
  def update_user_posts_counter
    author.update(posts_count: author.posts.count)
  end
```